### PR TITLE
fixed mysql syntax for multiple FROM and a JOIN

### PR DESCRIPTION
**Pull request by cebe**
fixes #6545

### CODE REVIEW

File: framework/db/mysql/QueryBuilder.php [hunk 1]
```diff
@@ -147,6 +147,18 @@ public function checkIntegrity($check = true, $schema = '', $table = '')
     /**
      * @inheritdoc
      */
+    public function buildFrom($tables, &$params)
+    {
+        if (count($tables) < 2) {
```
Comment (qiangxue) [inline framework/db/mysql/QueryBuilder.php]: shouldn't it also check if there's `join`? Because if there's no join, then we don't need the parenthesis.

Comment (samdark) [inline framework/db/mysql/QueryBuilder.php]: Parenthesis won't hurt in case of MySQL.

File: tests/unit/framework/db/QueryTest.php [hunk 2]
```diff
@@ -190,4 +190,21 @@ public function testOne()
...
     }
+
+    /**
+     * The combination of multiple tables in from and a join may cause problems in MySQL
+     * See https://github.com/yiisoft/yii2/pull/6545#issuecomment-67263973
+     */
+    public function testFromWithJoin()
+    {
+        if ($this->driverName === 'sqlite') {
+            return; // sqlite does not support this kind of syntax at all.
```
Comment (samdark) [inline tests/unit/framework/db/QueryTest.php]: Same about PostgreSQL. Such syntax isn't supported. It's MySQL only.

Comment (cebe) [inline tests/unit/framework/db/QueryTest.php]: I see, Postgres tests were broken/skipped

File: framework/db/mysql/QueryBuilder.php (hunk 1 above)
Comment (qiangxue) [inline framework/db/mysql/QueryBuilder.php]: While this is true, it just feels unnatural and unnecessary to have extra parenthesis. It's like you always add parenthesis to the SELECT part or other parts.

Comment (samdark) [inline framework/db/mysql/QueryBuilder.php]: Agree.

**Comment by samdark** [comment]
`PostgreSQLQueryTest::testFromWithJoin` fails.

**Comment by cebe** [comment]
I know, will fix it later

File: framework/db/mysql/QueryBuilder.php (hunk 1 above)
Comment (cebe) [inline framework/db/mysql/QueryBuilder.php]: `buildFrom` has no idea whether there is a join or not. If we want to add a check for that, this has to be done in `build()` but I do not see a good way for overriding that one.

**Comment by samdark** [comment]
Still fails after the fix: https://travis-ci.org/yiisoft/yii2/jobs/60672219#L194

**Comment by cebe** [comment]
was not a real fix, just rebased on master.

**Comment by cebe** [comment]
what about this one?

> `buildFrom()` has no idea whether there is a join or not. If we want to add a check for that, this has 
> to be done in `build()` but I do not see a good way for overriding that one.

**Comment by cebe** [comment]
https://github.com/yiisoft/yii2/pull/6545#issuecomment-263095898

**Pull request closed without merging**